scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2200,11 +2200,14 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val rpPos = auto.endTokenPos
       @inline def addPos(body: Term): Term = {
         body match {
-          // For `Term.Ref`, avoid using autoPos to exclude parens tokens in Term.ApplyInfix.args.
+          // For `Term.Ref` & `Lit` other than Unit, avoid using autoPos
+          // to exclude parens tokens in Term.ApplyInfix.args.
           // For `a f (b)`, autoPos include `(b)` for ApplyInfix.args position.
           // https://github.com/scalacenter/scalafix/issues/1594
           case _: Term.Ref =>
             body
+          case l: Lit if l.value != () =>
+            body
           case _ =>
             atPosWithBody(lpPos, body, rpPos)
```

Comment:
I am wondering if this should exist at all https://github.com/scalameta/scalameta/pull/2736/files#r862543397